### ISSUE TITLE
Modal disappear when opened second time

### ISSUE THREAD

**Issue by bruceCzK** [yuche/vue-strap#411]
Modal will be set `display:none` during animation when opened for second time.
Chrome  54.0.2840.98 (64-bit)

![image](https://cloud.githubusercontent.com/assets/8399711/20342162/49810838-ac24-11e6-85dc-b431e5b7e0d1.png)
![image](https://cloud.githubusercontent.com/assets/8399711/20342176/55828670-ac24-11e6-8697-00292c1276ac.png)

**Comment by bruceCzK** [comment]
I think it maybe related to [this change](https://github.com/yuche/vue-strap/commit/b128d870c5906c62c981e23ecffc0ddffbc947f8#diff-e3a681b966fcc1846ec5d2c8c6b4e494R118)

**Comment by mlaccetti** [comment]
Yeah, the `off` bit doesn't seem to be removing the event handlers properly, so the state gets mangled.

**Comment by isaacduane** [comment]
if (val) {
        $(el).find('.modal-content').focus();

        el.style.display = 'block';

        setTimeout(function () {
          $(el).addClass('in');
        }, 0);

        $(body).addClass('modal-open');

        if (scrollBarWidth !== 0) {
          body.style.paddingRight = scrollBarWidth + 'px'
        }

        if (this.backdrop) {
          $(el).on('click', function (e) {
            if (e.target === el) {
              that.show = false;
            }
          });
        }
      } else {
        body.style.paddingRight = null;

        $(body).removeClass('modal-open');

        $(el).removeClass('in');

        setTimeout(function () {
          $(el).off('click');

          el.style.display = 'none';
        }, 1000);
      }
